overlord/servicestate/quota_control.go

```diff
@@ -178,10 +171,38 @@ func RemoveQuota(st *state.State, name string) error {
 		return fmt.Errorf("cannot remove quota group with sub-groups, remove the sub-groups first")
 	}
 
+	// if this group has a parent, we need to remove the linkage to this
+	// sub-group from the parent first
+	if grp.ParentGroup != "" {
+		parent, ok := allGrps[grp.ParentGroup]
+		if !ok {
+			return fmt.Errorf("internal error: parent group of %q, %q not found", name, grp.ParentGroup)
+		}
+
+		newSubgroups := make([]string, 0, len(parent.SubGroups)-1)
```

Comment:
If the group in question is the only child of this parent, then I think we should avoid creating a 0-lenght list here?

Reply:
I think this is related to @bboozzoo question below.

Reply:
see my comment below, but basically it's not an issue if we create a 0-length list, since we just don't append anything to that list and still end up with an empty list that we assign back to SubGroups of the parent

Reply:
this is true but setting it to nil is tidier and we try to do that where possible

Reply:
sorry I missed @pedronis comment before pushing here, do you want me to address this as well to make this case more obvious and assign to nil directly ?

Reply:
well I pushed it anyways hopefully folks think the new way of doing this is reasonable